common/utils/src/main/resources/error/error-conditions.json

```diff
@@ -3267,6 +3267,12 @@
     },
     "sqlState" : "42000"
   },
+  "INVALID_ZONE_OFFSET": {
+    "message": [
+      "<message>. ZoneOffset represents the time difference from UTC and must be in the range from -18:00 to +18:00."
```

Comment:
it's a bad practice to pass in the entire message. We can just say the zone offset is invalid, and we keep the original error as the causedBy of this error.

Reply:
Not sure why we need to preserve the caused by. Or is that merely for the stack trace?

Reply:
yea, and also to retain the original error message from the JDK library.

Reply:
Applied comments and also updated corresponding constructor of `SparkDateTimeException`. Thanks!